modules/Contacts/views/view.detail.php

```diff
@@ -53,6 +53,17 @@ class ContactsViewDetail extends ViewDetail
 	public function display(){
 		global $sugar_config;
 
+		// Fix for issue 1506: removed picture is still cached in the Detail View.
```

Comment:
I agree. Besides its not only images - there are also problems with 
Firefox and Chrome caching language strings for dropdowns.

This bit is too specific:
         // Fix for issue 1506: removed picture is still cached in the 
Detail View.
         $js=<<<EOQ
         <script type='text/javascript'>
             if(!window.location.hash) {
                 window.location = window.location + '#loaded';
                 window.location.reload();
             }
         </script>
EOQ;
         echo $js;

for:

issue 2: Firefox and Google Chrome display cached image.
Steps to reproduce:
use image field in the Contacts module.
Add an image in Edit View, then save.
Go back to the edit view and and add another image and save again.
The detail view still displays the first image.



But this bit at least worked for
issue 1: IE11 does not display image saved through the field of type Image.

// Fix for issue 1506: IE11 cannot display generic 
'application/octet-stream' (which is defined as "arbitrary binary data" 
in RFC 2046).
$mime_type = mime_content_type($local_location);
if($mime_type == null || $mime_type == '') {
     $mime_type = 'application/octet-stream';
}

I should've created two pull requests as these are two different problems.

Do you want me create a new pull request with just:
// Fix for issue 1506: IE11 cannot display generic 
'application/octet-stream' (which is defined as "arbitrary binary data" 
in RFC 2046).
$mime_type = mime_content_type($local_location);
if($mime_type == null || $mime_type == '') {
     $mime_type = 'application/octet-stream';
}

- I tested it works for IE11.

On 27/05/16 15:42, Matt Lorimer wrote:
>
> In modules/Contacts/views/view.detail.php 
> <https://github.com/salesagility/SuiteCRM/pull/1507#discussion_r64915039>:
>
> > @@ -53,6 +53,17 @@ class ContactsViewDetail extends ViewDetail
> >  	public function display(){
> >  		global $sugar_config;
> >
> > +		// Fix for issue 1506: removed picture is still cached in the Detail View.
>
> this will only work for contacts and may not be the best approach to 
> solving this issue, I think a global solution for images is needed
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/salesagility/SuiteCRM/pull/1507/files/0eb14d576cee0eaeaed825b325d58e178287585e#r64915039>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/ANUZRGCRYyqm2UtoavAfcPovqd-ipvT0ks5qFwLFgaJpZM4Ioj5i>.
>